zabbix_cli/cli.py

```diff
@@ -6784,32 +6848,32 @@ def do_load_balance_proxy_hosts(self, args):
             self.generate_feedback('Error', 'Problems getting affected hosts')
             return False
 
-        #
-        # Create a dicctionary with hostid:proxyid entries. The
-        # proxyid value will be chosen randomly from the list of
-        # defined proxies.
-        #
+        # Select a proxy using the given weights.
 
         for hostid in all_hosts:
-            host_proxy_relation[hostid] = proxyid_list[random.randint(0, len(proxyid_list) - 1)]
+            host_proxy_relation[hostid] = random.choices(proxy_list, weight_list, k=1)[0]
 
         try:
-            for proxyid in proxyid_list:
+            for proxy in proxy_list:
                 hostid_list = []
 
-                for hostid, proxyid2 in host_proxy_relation.items():
-                    if proxyid2 == proxyid:
+                for hostid, chosen_proxy in host_proxy_relation.items():
+                    if chosen_proxy == proxy:
                         hostid_list.append({"hostid": str(hostid)})
+                        proxy_specs[proxy]['count'] += 1
 
                 query = {
                     "hosts": hostid_list,
-                    "proxy_hostid": proxyid
+                    "proxy_hostid": proxy_specs[proxy]['id']
                 }
 
                 result = self.zapi.host.massupdate(**query)
 
-            logger.info('Balanced configuration of hosts along defined proxies done')
-            self.generate_feedback('Done', 'Balanced configuration of hosts along defined proxies done')
+            proxy_distribution_str = ", ".join([f"{p}: {proxy_specs[p]['count']}"
+                                                for p in proxy_list])
+            logger.info('Balanced configuration of hosts along given proxies done: %s',
+                        proxy_distribution_str)
+            self.generate_feedback('Done', 'Balanced configuration of hosts along given proxies: ' + proxy_distribution_str)
```

Comment:
Same thing here: consider "among" instead of "along". Potentially "among _the_", but I can understand its omission due to brevity.